src/SwiftUI/SwiftUI/Color.cs

```diff
@@ -75,6 +66,59 @@ public Color (double hue, double saturation, double brightness, double opacity)
 		{
 			Data = CreateFromHSBO (hue, saturation, brightness, opacity);
 		}
+
+		public Color (RGBColorSpace colorSpace, double red, double green, double blue, double opacity)
+		{
+			var opaqueRBGColorspaceMetadata = SwiftType.Of (typeof (RGBColorSpace))!;
+			var result = Marshal.AllocHGlobal (opaqueRBGColorspaceMetadata.NativeDataSize);
+			try {
+				Data = CreateFromRGBColorSpaceRedGreenBlueOpacity (GetSwiftUIColorSpace (colorSpace, result.ToPointer()), red, green, blue, opacity);
+			} catch {
+				Marshal.FreeHGlobal (result);
+				throw;
+			}
+		}
+
+		public Color (RGBColorSpace colorSpace, double white, double opacity)
+		{
+			var opaqueRBGColorspaceMetadata = SwiftType.Of (typeof (RGBColorSpace))!;
+			var result = Marshal.AllocHGlobal (opaqueRBGColorspaceMetadata.NativeDataSize);
+			try {
+				Data = CreateFromRGBColorSpaceWhiteOpacity (GetSwiftUIColorSpace (colorSpace, result.ToPointer ()), white, opacity);
+			} catch {
+				Marshal.FreeHGlobal (result);
+				throw;
+			}
+		}
+
+		static void* GetSwiftUIColorSpace (RGBColorSpace colorSpace, void* result)
+		{
+			return colorSpace switch
+			{
+				RGBColorSpace.DisplayP3 => ColorSpaceDisplayP3 (result),
+				RGBColorSpace.sRGB => ColorSpacesRGB (result),
+				RGBColorSpace.sRGBLinear => ColorSpacesRGBLinear (result),
+				_ => throw new NotSupportedException (),
+			};
+		}
+
+		static void* ColorSpaceDisplayP3 (void*  result)
+		{
+			GetRGBColorSpaceDisplayP3 (result);
+			return result;
+		}
+
+		static void* ColorSpacesRGB (void* result)
+		{
+			GetRGBColorSpacesRGB (result);
+			return result;
+		}
+
+		static void* ColorSpacesRGBLinear (void* result)
+		{
+			GetRGBColorSpacesRGBLinear (result);
+			return result;
+		}
```

Comment:
Ah I see what's going on here.. we need to pass the return buffer for these. So perhaps it would be simpler to just use a regular switch statement, get rid of all these helper methods, and just have `GetSwiftUIColorSpace` return void?

Reply:
As we are just passing pointer around, we certainly can. I just made it fit this way based on your suggestion to use the new fangled switch.